ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Envelope.scala

```diff
@@ -52,8 +52,14 @@ object Envelope {
     enclose(Proto.Envelope.MessageKind.STATE_VALUE, stateValue.toByteString, compression)
 
   def open(envelopeBytes: ByteString): Either[String, Message] =
+    openWithParser(() => Proto.Envelope.parseFrom(envelopeBytes))
+
+  def open(envelopeBytes: Array[Byte]): Either[String, Message] =
+    openWithParser(() => Proto.Envelope.parseFrom(envelopeBytes))
+
+  private def openWithParser(parseEnvelope: () => Proto.Envelope): Either[String, Message] =
     for {
-      envelope <- Try(Proto.Envelope.parseFrom(envelopeBytes)).toEither.left.map(_.getMessage)
+      envelope <- Try(parseEnvelope()).toEither.left.map(_.getMessage)
```

Comment:
Perhaps you could pull the `Try(…).toEither.left.map(_.getMessage)` pattern into a helper.

Reply:
Changed accordingly.